src/middleware/checkPermissionMiddleware.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+
+interface User {
+  role: string;
+}
+
+interface CustomRequest extends Request {
+  user?: User;
+}
+
+const checkPermission =
+  (role: string) => (req: CustomRequest, res: Response, next: NextFunction) => {
+    if (req.user?.role === role) {
+      return next();
+    }
+    return res.status(401).json({ message: 'Not authorised, Not Seller' });
```

Comment:
This message would be inappropriate if I wanted to check if the logged-in user is an admin

Reply:
Updated. We are using dynamic variable in message generation 

<img width="451" alt="image" src="https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn/assets/97307891/81f4398c-a511-43ae-925e-b3504e455414">